filebeat/cmd/multiline/reader.go

```diff
@@ -0,0 +1,108 @@
+package multiline
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+	"github.com/elastic/beats/filebeat/harvester/reader"
+)
+
+type concatReader struct {
+	current io.ReadCloser
+	idx     int
+	paths   []string
+}
+
+func createReader(paths []string) (io.ReadCloser, error) {
+	switch len(paths) {
+	case 0:
+		return ioutil.NopCloser(os.Stdin), nil
+	case 1:
+		return os.Open(paths[0])
+	}
+	return newConcatReader(paths)
+}
+
+func createPipeline(config *prospectorConfig, in io.Reader) (reader.Reader, error) {
```

Comment:
I'm hoping in the future that we can just "reuse" this code from filebeat without having to reimplement that part.